include/boost/winasio/http/temp.hpp

```diff
@@ -79,14 +79,21 @@ class http_connection
                 << "async_recieve_request failed: " << ec.message();
           } else {
             self->on_receive_request();
-            // start another connection
-            std::make_shared<http_connection>(self->queue_handle_,
```

Comment:
I don't think this is right. 2 connections should run in parallel. If you use the same buffer (using reset) the data will be corrupted.

Reply:
The connection is like the session in beast example: https://www.boost.org/doc/libs/1_70_0/libs/beast/example/http/server/async/http_server_async.cpp

Reply:
How would they run in parallel, isn't this the completion handler? I will look at the example.